h2o-persist-gcs/src/main/java/water/persist/PersistGcs.java

```diff
@@ -0,0 +1,161 @@
+package water.persist;
+
+import com.google.cloud.ReadChannel;
+import com.google.cloud.storage.*;
+import water.H2O;
+import water.Key;
+import water.Value;
+import water.api.FSIOException;
+import water.fvec.GcsFileVec;
+import water.util.Log;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Persistence backend for GCS
+ */
+public final class PersistGcs extends Persist {
+
+  private static final Storage storage = StorageOptions.getDefaultInstance().getService();
+
+  @Override
+  public byte[] load(Value v) throws IOException {
+    final BlobId blobId = GcsBlob.of(v._key).getBlobId();
+    Log.info("Loading: " + blobId.toString());
```

Comment:
Our Log.info() will concant multiple objects automatically. A trivial issue. As the concat operation happens only once, performance is not affected ?